org.ultimatepp.TheIDE.yml

```diff
@@ -0,0 +1,115 @@
+    sources:
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/archive/refs/tags/2023.2.tar.gz
+        sha256: 8e0f02d2cdcd23f7774e9c576a8cc7afa50b97232046d09940e386cb89dcf286
+      - type: file
+        path: run-ide-install-host-deps.sh
+      - type: file
+        path: flatpak.patch
+  
+  - name: upp-term
+    buildsystem: simple
+    build-commands:
+      - mv 3p/Terminal/PtyProcess app/PtyProcess
+      - mv 3p/Terminal/PtyAgent app/PtyAgent
+      - mv 3p/Terminal/Terminal app/Terminal
+      - 3p/umk/umk app/,3p/uppsrc UppTerm 3p/umk/GCC.bm -brv +GUI,SHARED upp-term
+      - install -D upp-term /app/bin/upp-term
+    sources:
+      - type: archive
+        url: https://github.com/ultimatepp/UppTerm/archive/refs/tags/v2023.1.tar.gz
+        sha256: a40da7148d16f8f7c9989da1ad430d69c09483c91dab0607bc422ae6c184dda2
+      - type: archive
+        url: https://github.com/ismail-yilmaz/Terminal/archive/refs/tags/2023.2.tar.gz
+        sha256: 320d8574842825d1d8ddfa46addcb4eb03a9120116caea192582f2a02d26ec09
+        dest: 3p/Terminal
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/uppsrc-17045.tar.gz
+        sha256: 23793a363a140139728a30cc050c35287d6ff97ee03125d62f8ad8acf06e40af
+        dest: 3p/uppsrc
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/umk-17045-linux-x86-64.tar.gz
+        sha256: 36ee1a061590f73adf6367ef85c51adbfff164dd53a53480ae82eb0b9925b923
+        only-arches: [x86_64]
+        dest: 3p/umk
+      - type: archive
+        url: https://github.com/ultimatepp/ultimatepp/releases/download/2023.2/umk-17045-linux-aarch64.tar.gz
+        sha256: 2943ee5e044f3ff41393e921188b8b5a89cfca3ecaf1bfef66afb3a65882766c
+        only-arches: [aarch64]
+        dest: 3p/umk
+
+  - name: run-ide
+    buildsystem: simple
+    build-commands:
+      - install -D run-ide.sh /app/bin/run-ide.sh
+      - install -D TheIDE_512x512.png /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -D TheIDE_256x256.png /app/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -D TheIDE_128x128.png /app/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png
+      - install -D TheIDE_64x64.png /app/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - install -D org.ultimatepp.TheIDE.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - install -D org.ultimatepp.TheIDE.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: file
+        path: run-ide.sh
+      - type: file
+        path: org.ultimatepp.TheIDE.desktop
+      - type: file
+        path: org.ultimatepp.TheIDE.metainfo.xml
+      - type: file
+        path: res/TheIDE_512x512.png
+      - type: file
+        path: res/TheIDE_256x256.png
+      - type: file
+        path: res/TheIDE_128x128.png
+      - type: file
+        path: res/TheIDE_64x64.png
```

Comment:
only ONE icon since they don't come from upstream.

Reply:
no icon upstream?

Reply:
you can also submit a patc upstream for decently sized icon, and a SVG.

Reply:
Patch has been submitted to the upstream (https://github.com/ultimatepp/ultimatepp/pull/181) and accepted. Since, I can not use directly what is in the upstream due to lack of new release. I created simple patch from this PR (patch/icon.patch). Once the new version will be release it won't be needed.